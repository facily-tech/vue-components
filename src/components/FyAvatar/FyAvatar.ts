/// <reference path="../../../node_modules/vuetify/src/globals.d.ts" />
import Vue from 'vue';

import { VAvatar } from 'vuetify/lib';

const BaseVAvatar = Vue.extend({ mixins: [VAvatar] });

type options = InstanceType<typeof BaseVAvatar>;

export default BaseVAvatar.extend<options>().extend({
  name: 'fy-avatar',
  computed: {
    class() {
      return {
        ...VAvatar.options.computed.classes.call(this),
        'fy-avatar': true,
      };
    },
  },
});
