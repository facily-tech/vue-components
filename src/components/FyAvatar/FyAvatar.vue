<template>
  <v-avatar v-bind="$attrs" v-on="$listeners">
    <slot v-for="(_, name) in $slots" :name="name" :slot="name" />
  </v-avatar>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'FyAvatar',
});
</script>
