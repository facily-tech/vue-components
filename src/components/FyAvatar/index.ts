import FyAvatar from './FyAvatar.vue';

export { FyAvatar };

export default FyAvatar;
