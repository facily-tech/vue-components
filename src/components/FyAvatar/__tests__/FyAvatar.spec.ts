import Vue from 'vue';
import Vuetify from 'vuetify';
import FyAvatar from '../FyAvatar.vue';
import { mount, MountOptions, Wrapper } from '@vue/test-utils';

Vue.use(Vuetify);

describe('FyAvatar', () => {
  // eslint-disable-line max-statements
  type Instance = InstanceType<typeof FyAvatar>;
  let mountFunction: (options?: MountOptions<Instance>) => Wrapper<Instance>;
  beforeEach(() => {
    mountFunction = (options?: MountOptions<Instance>) => {
      return mount(FyAvatar, {
        vuetify: new Vuetify(),
        ...options,
      });
    };
  });

  it('should render component and match snapshot', () => {
    const wrapper = mountFunction();

    expect(wrapper.html()).toMatchSnapshot();
  });
});
