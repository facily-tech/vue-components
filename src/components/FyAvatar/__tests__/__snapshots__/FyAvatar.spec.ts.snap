// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FyAvatar should render component and match snapshot 1`] = `<div class="v-avatar" style="height: 48px; min-width: 48px; width: 48px;"></div>`;
